official/utils/misc/distribution_utils.py

```diff
@@ -30,15 +31,25 @@ def get_distribution_strategy(num_gpus, all_reduce_alg=None):
       See tf.contrib.distribute.AllReduceCrossDeviceOps for available
       algorithms. If None, DistributionStrategy will choose based on device
       topology.
+    use_one_device_strategy: Should only be set to Truen when num_gpus is 1.
+      If True, then use OneDeviceStrategy; otherwise, do not use any
+      distribution strategy.
 
   Returns:
     tf.contrib.distribute.DistibutionStrategy object.
   """
-  if num_gpus == 0:
+  if num_gpus == 0 and use_one_device_strategy:
     return tf.contrib.distribute.OneDeviceStrategy("device:CPU:0")
-  elif num_gpus == 1:
+  elif num_gpus == 0:
+    return None
+  elif num_gpus == 1 and use_one_device_strategy:
```

Comment:
I feel this should just call `tf.contrib.distribute.MirroredStrategy(num_gpus=num_gpus)` so it shows users you don't have to think about this it will just work.  Then the flag is `turn_off_distr_strat` and defaults False.

Reply:
great point. when this code was written, we couldn't use MirroredStrategy for 1 GPU. But I think now we can. So we can simplify this.

My question is: by default, if the user specifies only 0 or 1 GPU, do we want to use mirroredstrategy? or just not use any strategy?

Reply:
I would suggest always using a strategy (except for the `turn_off_distr_strat` case) for two reasons:
1) It does sensible optimizations (like prefetch_to_device), and I think the trend of using dist strat to always include safe best practices is a good one.
2) It can be confusing to debug issues otherwise. If a strategy is used with 2 GPUs but not 1, debugging issues is really confusing unless you know the code is doing that. And it would also make scalability numbers weird because the reference (1 GPU) is fundamentally different from the rest.

Reply:
sgtm. i think we can just use `MirroredStrategy` in all cases then, unless `turn_off_distr_strat` is set to true.